systems/console-bff/subscriptions/resolvers/resolver.ts

```diff
@@ -335,271 +335,329 @@ class SubscriptionsResolvers {
+      value: formatKPIValue(res.type, res.value),
+    }));
+  }
 
-            for (const key of metricKeys) {
-              metrics.metrics.push({
-                msg: "Error",
-                type: key,
-                siteId: siteId || "",
-                nodeId: nodeId || "",
-                success: false,
-                value: 0,
-              });
-            }
+  private setupSiteStateWSWorkers(
+    wsUrl: string,
+    metricKeys: string[],
+    data: GetMetricsSiteStatInput,
+    siteIds: string[],
+    nodeIds: string[] | undefined,
+    baseTopic: string
+  ) {
+    const handleSiteWSMessage = (
+      wsData: any,
+      topic: string,
+      siteId: string,
+      nodeId: string = ""
+    ) => {
+      if (!wsData.isError) {
+        try {
+          const res = JSON.parse(wsData.data);
+          if (res?.data?.result && res.data.result.length > 0) {
+            res.data.result.forEach((result: any) => {
+              if (
+                result &&
+                result.metric &&
+                result.value &&
+                result.value.length > 0
+              ) {
+                if (!nodeId) {
+                  const resultSiteId =
+                    result.metric.site || result.metric.instance || siteId;
+
+                  pubSub.publish(topic, {
+                    success: true,
+                    msg: "success",
+                    type: res.Name,
+                    siteId: resultSiteId,
+                    nodeId: "",
+                    value: [
+                      Math.floor(result.value[0]) * 1000,
+                      formatKPIValue(res.Name, result.value[1]),
+                    ],
+                  });
+                } else {
+                  const resultNodeId =
+                    result.metric.node || result.metric.nodeid || nodeId;
+
+                  pubSub.publish(topic, {
+                    success: true,
+                    msg: "success",
+                    type: res.Name,
+                    siteId: siteId,
+                    nodeId: resultNodeId,
+                    value: [
+                      Math.floor(result.value[0]) * 1000,
+                      formatKPIValue(res.Name, result.value[1]),
+                    ],
+                  });
+                }
+              }
+            });
           }
+        } catch (error) {
+          logger.error(
+            `Failed to parse WebSocket message for ${
+              nodeId ? `node ${nodeId}` : `site ${siteId}`
+            }: ${error}`
+          );
         }
       }
-    }
+    };
 
-    if (withSubscription && metrics.metrics.length > 0) {
-      const baseTopic = `stat-${data.orgName}-${userId}-${type}-${from}`;
+    for (const siteId of siteIds) {
+      const siteUrlParams = new URLSearchParams();
+      siteUrlParams.append("interval", data.step.toString());
+      siteUrlParams.append("metric", metricKeys.join(","));
+      siteUrlParams.append("site", siteId);
```

Comment:
Did you tried passing nodeId too in params to fetch site and node KPIs data with same thread?
Because currently you're doing it one by one, first fetching for site, then fetching for node